packages/sdk/src/services/PlatfformManager/openDeeplink.ts

```diff
@@ -38,14 +37,27 @@ export function openDeeplink(
 
     // It should only open after we can acknowledge that the rpc call that triggered the deeplink has been sent
     // TODO how can we know that the rpc call has been sent?
-    if (typeof window !== 'undefined') {
-      let win: Window | null;
-      if (state.useDeeplink) {
-        win = window.open(deeplink, '_blank');
-      } else {
-        win = window.open(universalLink, '_blank');
+    if (state.useDeeplink) {
+      if (typeof window !== 'undefined') {
+        // We don't need to open a deeplink in a new tab
+        // It avoid the browser to display a blank page
+        window.location.href = deeplink;
       }
-      setTimeout(() => win?.close?.(), LINK_OPEN_DELAY);
+    } else {
+      // Workaround for https://github.com/rainbow-me/rainbowkit/issues/524.
+      // Using 'window.open' causes issues on iOS in non-Safari browsers and
+      // WebViews where a blank tab is left behind after connecting.
+      // This is especially bad in some WebView scenarios (e.g. following a
+      // link from Twitter) where the user doesn't have any mechanism for
+      // closing the blank tab.
+      // For whatever reason, links with a target of "_blank" don't suffer
+      // from this problem, and programmatically clicking a detached link
+      // element with the same attributes also avoids the issue.
+      const link = document.createElement("a");
+      link.href = universalLink;
+      link.target = "_self";
+      link.rel = "noreferrer noopener";
+      link.click();
```

Comment:
should we be deleting this `link` element at some point? This seems like it would leave several links in the dom

Reply:
As we don’t add it to the DOM, this element will be garbage collected.